interfaces/input_parameters/ring_section.py

```diff
@@ -177,98 +165,134 @@ def __init__(self, section_length, alpha_0,
 
         # Reshaping the input synchronous data to the adequate format and
         # get back the momentum program
-        if not isinstance(synchronous_data, dTypes._ring_program):
+        if not isinstance(synchronous_data,
+                          ring_programs._synchronous_data_program):
             synchronous_data \
-                = dTypes._ring_program.conversions[func_type](synchronous_data)
+                = ring_programs._synchronous_data_program._conversions[
+                    func_type](synchronous_data)
```

Comment:
```suggestion
                = getattr(ring_programs, func_type)(synchronous_data)
```

Reply:
This didn't work, realized that I need to add '_program' to the func_type as
```python
getattr(ring_programs, func_type+'_program')(synchronous_data)
```
but not sufficient as total energy is defined in func_type as 'energy' but 'tot_energy_program' in the datatypes. Will keep as is at the moment.

Reply:
Bah!  Some uniformity in naming is needed, I propose 'total_energy' (or 'tot_energy') for the keyword and the same but with '_program' afterwards for the datatype.  What do you think?

Reply:
it's ok I forgot the
```python
        syncDataTypes = ('momentum', 'kinetic_energy', 'total_energy',
                         'bending_field')
```
was purely internal and not related to the \_\_init\_\_ kwargs, I renamed the syncDataTypes and used 
```python
getattr(ring_programs, func_type+'_program')(synchronous_data)
```

Reply:
D'Oh, should've noticed that!